plugin/src/main/java/io/jenkins/plugins/redis/RedisFingerprintStorage.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2020, Sumit Sarin and Jenkins project contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package io.jenkins.plugins.redis;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import hudson.Extension;
+import hudson.ExtensionList;
+import jenkins.fingerprints.FingerprintStorage;
+import hudson.model.Fingerprint;
+import hudson.Util;
+
+import java.io.IOException;
+import java.io.ByteArrayInputStream;
+import java.io.StringWriter;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+
+import jenkins.model.GlobalConfiguration;
+import org.jenkinsci.main.modules.instance_identity.InstanceIdentity;
+
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.JedisPoolConfig;
+import redis.clients.jedis.exceptions.JedisException;
+
+/**
+ * Pluggable external fingerprint storage for fingerprints in Redis.
+ */
+
+@Extension
+public class RedisFingerprintStorage extends FingerprintStorage {
+
+    private final String instanceId;
+    private static final Logger logger = Logger.getLogger(Fingerprint.class.getName());
```

Comment:
Oracle naming conventions recommend using uppercase for static final variable names